packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -125,7 +125,8 @@ class MouseTracker extends ChangeNotifier {
     final _TrackedAnnotation trackedAnnotation = _findAnnotation(annotation);
     for (int deviceId in trackedAnnotation.activeDevices) {
       if (annotation.onExit != null) {
-        annotation.onExit(PointerExitEvent.fromMouseEvent(_lastMouseEvent[deviceId]));
+        final PointerEvent event = _lastMouseEvent[deviceId] ?? _pendingRemovals[deviceId];
+        annotation.onExit(PointerExitEvent.fromMouseEvent(event));
```

Comment:
Could we add an `assert(event != null)` before this line? (Or are there cases where this is still legally null?)

Reply:
Done.